plugins/performance-lab/includes/admin/local-plugin-fallback.php

```diff
@@ -0,0 +1,225 @@
+		$fallback_data = array();
+
+		foreach ( $plugin_slugs as $plugin_slug ) {
+			// Look for plugin files that match this slug.
+			$plugin_file = perflab_find_local_plugin_file( $local_plugins, $plugin_slug );
+
+			if ( false === $plugin_file ) {
+				continue; // Plugin not installed locally.
+			}
+
+			$plugin_headers = $local_plugins[ $plugin_file ];
+			$is_active      = is_plugin_active( $plugin_file );
+
+			// Build normalized plugin data similar to WordPress.org API response.
+			$readme_description = perflab_get_plugin_readme_description( $plugin_file );
+			$description        = '' !== $readme_description ? $readme_description : ( $plugin_headers['Description'] ?? '' );
+
+			$fallback_data[ $plugin_slug ] = array(
+				'name'              => $plugin_headers['Name'] ?? $plugin_slug,
+				'slug'              => $plugin_slug,
+				'short_description' => $description,
+				'requires'          => $plugin_headers['RequiresWP'] ?? false,
+				'requires_php'      => $plugin_headers['RequiresPHP'] ?? false,
+				'requires_plugins'  => perflab_parse_requires_plugins( $plugin_headers, $plugin_slug ),
+				'version'           => $plugin_headers['Version'] ?? '0.0.0',
+				'is_installed'      => true,
+				'is_active'         => $is_active,
+			);
+		}
+
+		return $fallback_data;
+	}
+
+	/**
+	 * Finds the plugin file for a given slug among installed plugins.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array<string, array<string, string>> $local_plugins Array from get_plugins().
+	 * @param string                               $plugin_slug   Plugin slug to find.
+	 * @return string|false Plugin file path relative to plugins directory, or false if not found.
+	 */
+	function perflab_find_local_plugin_file( array $local_plugins, string $plugin_slug ) {
+		foreach ( $local_plugins as $plugin_file => $plugin_data ) {
+			// Extract directory name from plugin file path.
+			$plugin_dir = strtok( $plugin_file, '/' );
+
+			if ( $plugin_dir === $plugin_slug ) {
+				return $plugin_file;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Gets plugin description from readme file.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_file Plugin file path.
+	 * @return string Plugin description from readme or empty string.
+	 */
+	function perflab_get_plugin_readme_description( string $plugin_file ): string {
+		if ( ! defined( 'WP_PLUGIN_DIR' ) ) {
+			return '';
+		}
+
+		$plugin_dir   = dirname( WP_PLUGIN_DIR . '/' . $plugin_file );
+		$readme_files = array( 'readme.txt', 'README.txt', 'readme.md', 'README.md' );
+
+		foreach ( $readme_files as $readme_file ) {
+			$readme_path = $plugin_dir . '/' . $readme_file;
+			if ( file_exists( $readme_path ) ) {
+				$readme_content = file_get_contents( $readme_path );
+				if ( false !== $readme_content ) {
+					// Parse description from readme - look for description after "== Description ==".
+					if ( 1 === preg_match( '/==\s*Description\s*==(.*?)(?==|\z)/is', $readme_content, $matches ) ) {
```

Comment:
Is this regular expression right? The description is the paragraph _before_ the `== Description ==`. For example, it is line 10 here, not line 14:

https://github.com/WordPress/performance/blob/4f17dd77eede251e98fa7c95e73f7087ff0254c4/plugins/performance-lab/readme.txt#L1-L14